utilities/titandb/titan_db_test.cc

```diff
@@ -237,19 +300,67 @@ TEST_F(TitanDBTest, DbIter) {
 }
 
 TEST_F(TitanDBTest, Snapshot) {
-    Open();
-    std::map<std::string, std::string> data;
-    Put(1, &data);
-    ASSERT_EQ(1, data.size());
+  Open();
+  std::map<std::string, std::string> data;
+  Put(1, &data);
+  ASSERT_EQ(1, data.size());
 
-    const Snapshot* snapshot(db_->GetSnapshot());
-    ReadOptions ropts;
-    ropts.snapshot = snapshot;
+  const Snapshot* snapshot(db_->GetSnapshot());
+  ReadOptions ropts;
+  ropts.snapshot = snapshot;
 
-    VerifyDB(data, ropts);
-    Flush();
-    VerifyDB(data, ropts);
-    db_->ReleaseSnapshot(snapshot);
+  VerifyDB(data, ropts);
+  Flush();
+  VerifyDB(data, ropts);
+  db_->ReleaseSnapshot(snapshot);
+}
+
+TEST_F(TitanDBTest, IngestExternalFiles) {
+  Open();
+  SstFileWriter sst_file_writer(EnvOptions(), options_);
+  ASSERT_EQ(sst_file_writer.FileSize(), 0);
+
+  const uint64_t kNumEntries = 100;
+  std::map<std::string, std::string> data;
+  for (uint64_t i = 1; i <= kNumEntries; i++) {
+    Put(i, &data);
+  }
+  ASSERT_EQ(kNumEntries, data.size());
+  VerifyDB(data);
+  Flush();
+  VerifyDB(data);
+
+  const uint64_t kNumIngestedEntries = 100;
+  // Make sure that keys in SST overlaps with existing keys
+  const uint64_t kIngestedStart = kNumEntries - kNumEntries / 2;
+  std::string sst_file = options_.dirname + "/for_ingest.sst";
+  ASSERT_OK(sst_file_writer.Open(sst_file));
+  for (uint64_t i = 1; i <= kNumIngestedEntries; i++) {
+    std::string key = GenKey(kIngestedStart + i);
+    std::string value = GenValue(kIngestedStart + i);
+    ASSERT_OK(sst_file_writer.Put(key, value));
+    auto iter = data.find(key);
+    if(iter != data.end()) {
+      iter->second = value;
+    } else {
+      data.emplace(key, value);
+    }
+  }
+  ASSERT_OK(sst_file_writer.Finish());
+  IngestExternalFileOptions ifo;
+  ASSERT_OK(db_->IngestExternalFile({sst_file}, ifo));
+  VerifyDB(data);
+  Flush();
+  VerifyDB(data);
+  for(auto& handle : cf_handles_) {
+    auto blob = GetBlobStorage(handle);
+    ASSERT_EQ(0, blob.lock()->NumBlobFiles());
```

Comment:
Sorry, I might have mislead you. The number of blob files should be 1 here, because before ingestion we also inserted some values with size > min_blob_size. Please check.